Updated Validators, cleaned some stuff up

smithal5 opened the pull request:
Updated validators for loginfo, test, and unit fields:
End date should follow start date
Test and Unit lower depths should be lower than upper depths

Added numeric keyboard to number only fields

~~Changed lat long to north east~~ (now have north east and lat long)

Cleaned up some field stuff, and dart analysis stuff

Changed -m to m

Added a notes field to unit with max length 256. Added max lengths to all other fields, text are 100, numbers are 15, north/east/lat/long are 25.
Added projection dropdown with ~~test values. Default is Lambert.~~ values from Victoria. Default is NAD83
Added depth overlap check to units and tests. Only checks when saved.

amayaevan commented on blamo/lib/Unit/unit.dart:
```diff
@@ -304,25 +313,23 @@ class _UnitPageState extends State<UnitPage> {
   }
 
-  void saveObject() async{
+  Future<void> saveUnitObject() async{
```
Changed this function name to saveUnitObject but you didn't update the function call on ln:274 thus causing a build error.

amayaevan commented on blamo/lib/Test/tests.dart:
```diff
@@ -222,56 +222,10 @@ class _TestsPageState extends State<TestsPage> {
   }
 
   void _onTileLongClicked(int i) async {
-    String result;
```
I think you missed this part in the merging. Removed the ability to delete a test

amayaevan commented on blamo/lib/main.dart:
```diff
@@ -195,76 +194,26 @@ class _HomePageState extends State<HomePage> {
-  }
-
   void _onTileLongPressed(int index) async {
-    currentState.documentIterator = index;
```
Same here. Didn't merge this in for the delete functionality.

amayaevan commented on blamo/lib/main.dart:
```diff
@@ -195,76 +194,26 @@ class _HomePageState extends State<HomePage> {
     //currentState.currentDocument = "";
   }
 
-  void _showToast(String toShow, MaterialColor color){
```
Missed merging this in too

amayaevan commented on blamo/lib/Test/test.dart:
```diff
@@ -150,21 +150,27 @@ class _TestPageState extends State<TestPage> {
                                 attribute: 'endTestDepth',
-                                validators: [FormBuilderValidators.numeric()],
+                                validators: [FormBuilderValidators.numeric(), FormBuilderValidators.max(0), (endDepth){
+                                  if(_fbKey.currentState != null && endDepth != null && _fbKey.currentState.fields["beginTestDepth"].currentState.value != null && double.tryParse(_fbKey.currentState.fields["beginTestDepth"].currentState.value) != null && double.tryParse(endDepth) != null && double.tryParse(endDepth) > double.tryParse(_fbKey.currentState.fields["beginTestDepth"].currentState.value ...
```
add >= to make the evaluation with the case of the values being equal when they should not be allowed to.

amayaevan commented on blamo/lib/Unit/unit.dart:
```diff
@@ -148,47 +150,54 @@ class _UnitPageState extends State<UnitPage> {
                                 attribute: 'depth-lb',
-                                validators: [FormBuilderValidators.numeric()],
+                                validators: [FormBuilderValidators.numeric(), FormBuilderValidators.max(0), (lower){
+                                  if(_fbKey.currentState != null && lower != null && _fbKey.currentState.fields['depth-ub'].currentState.value != null && double.tryParse(_fbKey.currentState.fields['depth-ub'].currentState.value) != null && double.tryParse(lower) != null && double.tryParse(lower) > double.tryParse(_fbKey.currentState.fields['depth-ub'].currentState.value))
```
same here >=

jctrotter commented on blamo/lib/Test/test.dart:
```diff
@@ -150,21 +150,27 @@ class _TestPageState extends State<TestPage> {
+                                    return "End Test Depth must be lower than Begin Test Depth";
+                                  return null;
+                                }],//Custom validator that checks that end depth is lower than begin depth
                                 decoration: InputDecoration(labelText: "End Test Depth (-m)"),
```
Should we remove the negative from the unit here so it'd read "(m)?" Currently (-m) could imply that whatever the user inputs will be made negative after the fact, which isn't the case.

jctrotter commented on blamo/lib/Test/test.dart:
```diff
@@ -150,21 +150,27 @@ class _TestPageState extends State<TestPage> {
                                 attribute: 'beginTestDepth',
-                                validators: [FormBuilderValidators.numeric()],
+                                validators: [FormBuilderValidators.numeric(), FormBuilderValidators.max(0)],
                                 decoration: InputDecoration(labelText: "Begin Test Depth (-m)"),
```
Should we remove the negative from the unit here so it'd read "(m)?" Currently (-m) could imply that whatever the user inputs will be made negative after the fact, which isn't the case.

jctrotter commented on blamo/lib/Unit/unit.dart:
```diff
@@ -148,47 +150,54 @@ class _UnitPageState extends State<UnitPage> {
+                                    return "Depth Lower Bound must be lower than Depth Upper Bound";
+                                  return null;
+                                }],//Custom validator that checks if lower bound is lower than upper bound
                                 decoration: InputDecoration(labelText: "Depth Lower Bound (-m)"),
```
Should we remove the negative from the unit here so it'd read "(m)?" Currently (-m) could imply that whatever the user inputs will be made negative after the fact, which isn't the case.

smithal5 commented:
Added a notes field to unit with max length 256. Added max lengths to all other fields, text are 100, numbers are 15, north/east/lat/long are 25.
Added projection dropdown with test values. Default is Lambert.
Added depth overlap check to units and tests. Only checks when saved.

amayaevan commented on blamo/lib/Unit/unit.dart:
```diff
@@ -148,47 +150,74 @@ class _UnitPageState extends State<UnitPage> {
-                                decoration: InputDecoration(labelText: "Depth Lower Bound (-m)"),
+                                validators: [FormBuilderValidators.numeric(), FormBuilderValidators.max(0), FormBuilderValidators.required(), (lower){
+                                  if(_fbKey.currentState != null && lower != null && _fbKey.currentState.fields['depth-ub'].currentState.value != null && double.tryParse(_fbKey.currentState.fields['depth-ub'].currentState.value) != null && double.tryParse(lower) != null && double.tryParse(lower) >= double.tryParse(_fbKey.currentState.fields['depth-ub'].currentState.value))
+                                    return "Depth Lower Bound must be lower than Depth Upper Bound";
```
Error message is getting cut off here. ![image](https://user-images.githubusercontent.com/43584732/80058474-2f561c00-84de-11ea-876c-fc9cd919ea1a.png)

Probably just change the error string to "Lower Bound must be lower than Upper Bound"

amayaevan commented on blamo/lib/Unit/units.dart:
```diff
@@ -269,7 +268,7 @@ Widget getScaffold(List<Unit> units){
   }
 
-void testBuiildingList() async {
+  void testBuiildingList() async {
```
Is this function being used for anything? Can probably be removed or at least typo fixed in name.

PixelDotPixel commented on blamo/lib/Export/CSVExporter.dart:
```diff
@@ -71,8 +71,8 @@ class CSVExporter {
           "${tests[i].endTest},"
-          "${formatTags(tests[i].tags)},"
           ","
+          "${formatTags(tests[i].tags)},"
```
Add the new notes field somewhere in here

PixelDotPixel commented on blamo/lib/Unit/unit.dart:
```diff
@@ -282,6 +317,27 @@ class _UnitPageState extends State<UnitPage> {
+    for(int i = 0; i < currentState.unitList.length; i++){
+      Unit currentCheck = await objectHandler.getUnitData(currentState.unitList[i], currentState.currentDocument);
+      if(currentState.currentUnit != currentState.unitList[i]) {
+        if (unitObject.depthUB < currentCheck.depthUB && unitObject.depthUB > currentCheck.depthLB) {
```
Error thrown here if trying to save an existing test with modification, prevents the saving.

PixelDotPixel commented on blamo/lib/LogInfo/loginfo.dart:
```diff
@@ -215,68 +199,98 @@ class _LogInfoPageState extends State<LogInfoPage> {
-                                  FocusScope.of(context).requestFocus(formNodes[5]);
-                                },
+                              ),
+                              FormBuilderDropdown(
```
Projections breaks, returns null value somewhere along the way resulting in: 
![image](https://user-images.githubusercontent.com/43582537/80268141-e75c0400-8659-11ea-8305-e1be580177e7.png)
